hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -624,6 +627,10 @@ public String getReferencedPropertyName() {
 		return referencedPropertyName;
 	}
 
+	public boolean isReferenceToPrimaryKey() {
+		return referenceToPrimaryKey;
```

Comment:
Wouldn't it be enough to use `referencedPropertyName == null`? AFAIU this should always be null in case the primary key is referenced.

Reply:
I'm trying it

Reply:
From my tests you're right. Fixed it. Thanks!